test/e2e/set/scyllacluster/scyllacluster_replace.go

```diff
@@ -25,156 +21,119 @@ import (
+		targetServiceName := utils.GetNodeName(sc, 0)
+		targetService, err := nsClient.KubeClient().CoreV1().Services(sc.Namespace).Get(ctx, targetServiceName, metav1.GetOptions{})
 		o.Expect(err).NotTo(o.HaveOccurred())
 
-		preReplaceService := replacedNodeService.DeepCopy()
-		framework.Infof("Initial service %q UID is %q", preReplaceService.Name, preReplaceService.UID)
+		targetPod, err := nsClient.KubeClient().CoreV1().Pods(sc.Namespace).Get(ctx, targetServiceName, metav1.GetOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
 
-		framework.By("Replacing a node #0")
-		pod, err := f.KubeClient().CoreV1().Pods(f.Namespace()).Get(
-			ctx,
-			utils.GetNodeName(sc, 0),
-			metav1.GetOptions{},
-		)
+		preReplacementHostID, err := utils.GetHostID(ctx, nsClient.KubeClient().CoreV1(), sc, targetService, targetPod)
+		o.Expect(err).NotTo(o.HaveOccurred(), "Can't get the host ID of the node being replaced")
+
+		framework.By("Initiating the node replacement procedure")
+		targetServiceCopy := targetService.DeepCopy()
+		metav1.SetMetaDataLabel(&targetServiceCopy.ObjectMeta, naming.ReplaceLabel, "")
+
+		patch, err := controllerhelpers.GenerateMergePatch(targetService, targetServiceCopy)
 		o.Expect(err).NotTo(o.HaveOccurred())
-		framework.Infof("Initial pod %q UID is %q", pod.Name, pod.UID)
 
-		_, err = f.KubeClient().CoreV1().Services(f.Namespace()).Patch(
+		_, err = nsClient.KubeClient().CoreV1().Services(sc.Namespace).Patch(
 			ctx,
-			pod.Name,
+			targetService.Name,
 			types.MergePatchType,
-			[]byte(fmt.Sprintf(
-				`{"metadata":{"labels": {"%s": ""}}}`,
-				naming.ReplaceLabel,
-			)),
+			patch,
 			metav1.PatchOptions{},
 		)
 		o.Expect(err).NotTo(o.HaveOccurred())
 
-		waitCtx2, waitCtx2Cancel := utils.ContextForRollout(ctx, sc)
-		defer waitCtx2Cancel()
-
-		framework.By("Waiting for the pod to be replaced")
-		_, err = controllerhelpers.WaitForPodState(waitCtx2, f.KubeClient().CoreV1().Pods(pod.Namespace), pod.Name, controllerhelpers.WaitForStateOptions{TolerateDelete: true}, func(p *corev1.Pod) (bool, error) {
-			return p.UID != pod.UID, nil
-		})
-		o.Expect(err).NotTo(o.HaveOccurred())
-
-		// Give the controller some time to observe that the pod is down.
-		time.Sleep(10 * time.Second)
-
-		framework.By("Waiting for the ScyllaCluster to re-deploy")
-		waitCtx3, waitCtx3Cancel := utils.ContextForRollout(ctx, sc)
-		defer waitCtx3Cancel()
-		sc, err = controllerhelpers.WaitForScyllaClusterState(waitCtx3, f.ScyllaClient().ScyllaV1().ScyllaClusters(sc.Namespace), sc.Name, controllerhelpers.WaitForStateOptions{}, utils.IsScyllaClusterRolledOut)
-		o.Expect(err).NotTo(o.HaveOccurred())
-
-		scyllaclusterverification.Verify(ctx, f.KubeClient(), f.ScyllaClient(), sc)
+		var wg sync.WaitGroup
+		// Wait for the controllers to pick up the Pod replacement.
+		// Do this in the background to ensure we don't miss the progressing state if it happens too quickly.
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			g.GinkgoRecover()
```

Comment:
Shouldn't we defer instead?
```suggestion
			defer g.GinkgoRecover()
```